generic.go

```diff
@@ -58,6 +58,33 @@ func GenericContainer(ctx context.Context, req GenericContainerRequest) (Contain
 	return c, nil
 }
 
+// GenericReusableContainer reuses a container if it exists or creates a new one with parameters
+// Reusing will work only if you pass an existing container's name via 'req.Name' field
+// If the name is empty or it is not in a list of existing containers, the function will create a new generic container
+
+func GenericReusableContainer(ctx context.Context, req GenericContainerRequest) (Container, error) {
```

Comment:
Yeah, I agree with you. Could you suggest me, is it ok if I add this logic to `GenericContainer` ? Because `GenericReusableContainer ` and `GenericContainer` have similar input arguments and return values. 

If it is ok, i will add new comments and examples to 'GenericContainer'.

Reply:
Let me check how the Java flavour supports it. Will ping you back here